clang/lib/DPCT/CUBAPIMigration.cpp

```diff
@@ -132,19 +133,29 @@ void CubDeviceLevelRule::runRule(
 void CubMemberCallRule::registerMatcher(ast_matchers::MatchFinder &MF) {
   MF.addMatcher(
       cxxMemberCallExpr(
-          allOf(on(hasType(hasCanonicalType(qualType(hasDeclaration(
-                    namedDecl(hasName("cub::ArgIndexInputIterator"))))))),
-                callee(cxxMethodDecl(hasName("normalize")))))
-          .bind("ArgIndexInputIterator.normalize"),
+          anyOf(
+              allOf(on(hasType(hasCanonicalType(qualType(hasDeclaration(
+                        namedDecl(hasName("cub::ArgIndexInputIterator"))))))),
+                    callee(cxxMethodDecl(hasName("normalize")))),
+              callee(cxxMethodDecl(hasAnyName(
+                  "Current", "Alternate")))))
+          .bind("memberCall"),
+      this);
+
+  MF.addMatcher(
+      memberExpr(member(hasAnyName("d_buffers")))
+      .bind("memberExpr"),
       this);
 }
 
 void CubMemberCallRule::runRule(
     const ast_matchers::MatchFinder::MatchResult &Result) {
-  if (const auto *MC = getNodeAsType<CXXMemberCallExpr>(
-          Result, "ArgIndexInputIterator.normalize")) {
+  auto or_ = [](const Expr *x, const Expr *y) { return x ? x : y; };
+  if (const Expr *E = or_(
```

Comment:
How about use code like:

ExprAnalysis EA;
if (auto E1 = ...)
  EA.analyze(E1);
else if (auto E2 = ...)
  EA.analyze(E2);
...

It is more friendly for other people and expanding.